### PR TITLE
Add Virchow2 model to registry

### PR DESCRIPTION
**Pull request by nkaenzig**
Contributes to #694

### CODE REVIEW

File: src/eva/core/models/transforms/extract_cls_features.py [hunk 1]
```diff
@@ -7,13 +7,20 @@
 class ExtractCLSFeatures:
     """Extracts the CLS token from a ViT model output."""
 
-    def __init__(self, cls_index: int = 0) -> None:
+    def __init__(
+        self, cls_index: int = 0, num_reg_tokens: int = 0, include_patch_tokens: bool = False
```
Comment (roman807) [inline src/eva/core/models/transforms/extract_cls_features.py:11]: lets rename `num_reg_tokens` -> `num_register_tokens`? (reg could also mean something else)

File: src/eva/vision/models/networks/backbones/pathology/histai.py [hunk 2]
```diff
@@ -42,5 +42,5 @@ def histai_hibou_l(out_indices: int | Tuple[int, ...] | None = None) -> nn.Modul
         model_name="histai/hibou-L",
         out_indices=out_indices,
         model_kwargs={"trust_remote_code": True},
-        transform_args={"ignore_remaining_dims": True} if out_indices is not None else None,
+        transform_args={"num_reg_tokens": 4} if out_indices is not None else None,
```
Comment (roman807) [inline src/eva/vision/models/networks/backbones/pathology/histai.py:45]: maybe make this a variable, or explain in a comment where "4" is coming from?

File: src/eva/core/models/transforms/extract_cls_features.py (hunk 1 above)
Comment (nkaenzig) [reply src/eva/core/models/transforms/extract_cls_features.py]: done

File: src/eva/vision/models/networks/backbones/pathology/histai.py (hunk 2 above)
Comment (nkaenzig) [reply src/eva/vision/models/networks/backbones/pathology/histai.py]: This needs to be a constant, is a hyperparameter of the ViT (their models use 4 register tokens).

Comment (nkaenzig) [reply src/eva/vision/models/networks/backbones/pathology/histai.py]: Added a comment now to the docstring.

**Pull request merged**